Jenkinsfile

```
@@ -11,52 +15,115 @@ pipeline {
             }
         }
 
-        stage('Unit Tests') {
+        stage('Build') {
             steps {
-                timeout(30) {
-                    sh "./gradlew clean test --info --info --stacktrace --no-daemon"
+                script {
+                    // Also publish to local maven repo as this is required for nexus
+                    sh './gradlew assemble contracts:publishToMavenLocal workflows:publishToMavenLocal -Dmaven.repo.local="${MAVEN_LOCAL_PUBLISH}" --parallel'
+                    sh 'ls -lR "${MAVEN_LOCAL_PUBLISH}"'
                 }
             }
         }
 
-        stage('Integration Tests') {
+        stage('Nexus Scan') {
+            steps {
+                script {
+                    def props = readProperties file: 'gradle.properties'
+                    version = props['version']
+                    groupId = props['group']
+                    def artifactId = 'sdk-tokens'
+                    nexusAppId = "${groupId}-${artifactId}-${version}"
+                    echo nexusAppId
+                }
+                dir(mavenLocal) {
+                    script {
+                        fileToScan = findFiles(
+                            excludes: '**/*-javadoc.jar, **/*-sources.jar',
+                            glob: '**/*.jar, **/*.zip'
+                        ).collect {
+                            f -> [scanPattern: f.path]
+                        }
+                    }
+                    nexusPolicyEvaluation(
+                        failBuildOnNetworkError: true,
+                        iqApplication: nexusAppId, // application *has* to exist before a build starts!
+                        iqScanPatterns: fileToScan,
+                        iqStage: nexusIqStage()
+                    )
+                }
+            }
+        }
+
+        stage('Snyk Security') {
+            when {
+                expression { isReleaseTag() || isReleaseBranch() }
+            }
+            steps {
+                script {
+                    // Invoke Snyk for each Gradle sub project we wish to scan
+                    def modulesToScan = ['contracts', 'workflows']
+                    modulesToScan.each { module ->
+                        snykSecurityScan(env.SNYK_TOKEN, "--sub-project=$module --configuration-matching='^runtimeClasspath\$' --prune-repeated-subdependencies --debug --target-reference='${env.BRANCH_NAME}' --project-tags=Branch='${env.BRANCH_NAME.replaceAll("[^0-9|a-z|A-Z]+","_")}'")
+                    }
+                }
+            }
+        }
+        stage('Unit / Integration Tests') {
             steps {
                 timeout(30) {
-                    sh "./gradlew integrationTest --info --stacktrace --no-daemon"
+                    sh "./gradlew test integrationTest -Si --no-daemon --parallel"
                 }
             }
         }
 
         stage('Freighter Tests') {
+           when {
+                expression { params.DO_TEST }
```

Minor change required: 
```suggestion
                expression { params.RUN_FREIGHTER_TESTS}
```